### ISSUE TITLE
Add a recipe "source" attribute

### ISSUE THREAD

**Issue by Maschine2** [TandoorRecipes/recipes#2381]
### Describe the solution you'd like

Please add a new attribute "source" for recipes. For reference, I want to know where I have a certain recipe from. This could be a classic printed cookbook where the "source" would be something like "Basic Cooking, page 123" or a website URL which could point to a YouTube video, a foodblog, or a classic recipe page like chefkoch.

If the "source" attribute contains an URL, it should be converted to a clickable link in the recipe view and open the referenced source page in a new browser tab.

The new field could be added next to the recipe description.

### Describe alternatives you've considered

I've tried to add the source as part of the recipe description. But in this free text field, a structured approach to managing sources is not possible. Also, you might forget to add a source if there is not dedicated input field which reminds you to do so.

I think it's also not possible to click on a link in the description text, if the recipe source is a website. This would require copy/paste to view the referenced recipe source.

**Comment by vabene1111** [comment]
there is already a field called "source url" which can be used for urls, you could probably also just add a text into that field, maybe i will rename it from source url to just source.

**Comment by Maschine2** [comment]
Thanks for your reply. I think I was too focused on the term "source" and was misled by the German Translation "Importiert aus". The current implementation works well for valid URLs.

However, when adding text to the field, the app tries to generate a (non-exiting) URL from the text:
![grafik](https://user-images.githubusercontent.com/42979603/227644542-3852fc48-9f6b-4db7-b071-defd1001b1e2.png)
Could you please consider changing the behaviour to only create a link if the URL format is valid and display everything else as plain text?

**Comment by vabene1111** [comment]
yes that would make sense. I will probably also look at the translation so it makes more sense, maybe the source naming needs changing as well. Or we might actually need two fields, could be that its easier than checking if the field contains a url.

**Comment by supaeasy** [comment]
I suggest "Quelle" as the german translation.
Also: It would be great to have two fields: one for the link and one for a link description (or non-Internet-Sources like Cookbooks and Grannys) That creates a link only when there is a valid one in the first field. It just looks untidy to display a full link.
